cognite_toolkit/_cdf_tk/resource_classes/view_field_definitions.py

```diff
@@ -0,0 +1,208 @@
+import re
+import sys
+from types import MappingProxyType
+from typing import Any, ClassVar, Literal, cast
+
+from pydantic import Field, ModelWrapValidatorHandler, model_serializer, model_validator
+from pydantic_core.core_schema import SerializerFunctionWrapHandler
+
+from cognite_toolkit._cdf_tk.constants import (
+    CONTAINER_AND_VIEW_EXTERNAL_ID_PATTERN,
+    CONTAINER_AND_VIEW_PROPERTIES_IDENTIFIER_PATTERN,
+    SPACE_FORMAT_PATTERN,
+    VIEW_VERSION_PATTERN,
+)
+from cognite_toolkit._cdf_tk.utils.collection import humanize_collection
+
+from .base import BaseModelResource
+from .container_field_definitions import ContainerReference
+
+if sys.version_info < (3, 11):
+    from typing_extensions import Self
+else:
+    from typing import Self
+
+KEY_PATTERN = re.compile(CONTAINER_AND_VIEW_PROPERTIES_IDENTIFIER_PATTERN)
+
+
+class ViewReference(BaseModelResource):
+    type: Literal["view"] = "view"
+    space: str = Field(
+        description="Id of the space that the view belongs to.",
+        min_length=1,
+        max_length=43,
+        pattern=SPACE_FORMAT_PATTERN,
+    )
+    external_id: str = Field(
+        description="External-id of the view.",
+        min_length=1,
+        max_length=255,
+        pattern=CONTAINER_AND_VIEW_EXTERNAL_ID_PATTERN,
+    )
+    version: str = Field(
+        description="Version of the view.",
+        max_length=43,
+        pattern=VIEW_VERSION_PATTERN,
+    )
+
+
+class DirectRelationReference(BaseModelResource):
+    space: str = Field(
+        description="Id of the space that the view belongs to.",
+        min_length=1,
+        max_length=43,
+        pattern=SPACE_FORMAT_PATTERN,
+    )
+    external_id: str = Field(
+        description="External-id of the view.",
+        min_length=1,
+        max_length=255,
+        pattern=CONTAINER_AND_VIEW_EXTERNAL_ID_PATTERN,
+    )
+
+
+class ThroughRelationReference(BaseModelResource):
+    source: ViewReference | ContainerReference = Field(
+        description="Reference to the view or container from where this relation is inherited.",
+    )
+    identifier: str = Field(
+        description="Identifier of the relation in the source view or container.",
+        min_length=1,
+        max_length=255,
+    )
+
+
+class ViewProperty(BaseModelResource):
+    name: str | None = Field(
+        default=None,
+        description="Name of the property.",
+        max_length=255,
+    )
+    description: str | None = Field(
+        default=None,
+        description="Description of the content and suggested use for this property..",
+        max_length=1024,
+    )
+
+    @model_validator(mode="wrap")
+    @classmethod
+    def find_property_type_cls(cls, data: Any, handler: ModelWrapValidatorHandler[Self]) -> Self:
```

Comment:
I can't see any tests for this

Reply:
Here is the line by line coverage of this particular function:
![image](https://github.com/user-attachments/assets/dcb240f8-1372-4be2-861d-1a9e44a1c6fe)

Reply:
is there any way to get the coverage report without running it myself?

Reply:
Yes, green - hit, red not hit.

Reply:
So we have mostly happy-path testing